File: PRACTICAS/PRACTICA_ACTUAL/RUBY/lib/sorpresa_ircarcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Sorpresa_ircarcel < Sorpresa
    
    
    def initialize(n)
      @texto="VAS A LA CARCEL"
      @tablero = n
    end    
    public_class_method :new

    
    
    def aplicar_a_jugador(actual,todos)
      if actual>=0&&actual<todos.size
        informe(actual,todos)
        todos[actual].encarcelar(@tablero.attr_get_carcel)
      end
    end
    
    
  end
end

File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/civitas_p1/SorpresaIrCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.ArrayList;

/**
 *
 * @author angelsolano
 */

public class SorpresaIrCarcel extends Sorpresa{

    private Tablero tablero;
    
    SorpresaIrCarcel(Tablero tablero, String texto){
        super(texto);
        this.tablero = tablero;
        
    }

    @Override
     void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());   
        }
    }
    
    
    
}
